airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/error_handlers/default_error_handler.py

```diff
@@ -130,6 +131,11 @@ def max_retries(self, value: int) -> None:
     def interpret_response(self, response: requests.Response) -> ResponseStatus:
         request = response.request
 
+        if isinstance(request, CachedRequest):
```

Comment:
There are two reasons for this:
1. As you described, I don't think it's necessary to process the same request several times.
2. The request is used as a key to variable `self._last_request_to_attempt_count: MutableMapping[requests.PreparedRequest, int] = {}`, and while a regular request can be hashed, `CachedRequest` does not support the `__hash__` method

Reply:
I'd prefer we find a way to abstract this out because this forces all users to know the type of request that is passed. 

I'm not too concerned about the first case - I'm fine with re-processing requests in the error handler.

The second point is both interesting and somewhat problematic. I think a better solution would be to use our own Request struct instead of trying to hash the PreparedRequest.

It would look something like

```
@dataclass
class HttpRequest:
    method: HttpMethod
    url: str
    headers: Mapping[str, Any]
    body: Union[str, Mapping[str, Any]]
```

that way we don't need to worry about how the request is implemented and whether it supports hashing.

what do you think?

Reply:
I didn't quite understand exactly where to use own HttpRequest dataclass, so wanted to clarify.

Are you suggesting to replace the request (PreparedRequest or CachedRequest with own HttpRequest) in the response before passing the response to the `interpret_response(self, response: requests.Response)` method?

Reply:
Yes, that's what I was suggesting. That being said, I played a bit with the idea, and it's not as much of an easy win as I thought since we can't hash dataclass objects that have a dict either.

A good middle ground solution that won't derail us for too much would be to
- use `CachedRequest.prepare` to get the underlying PreparedRequest
- always go through interpret_response
- add a check to ensure the request passed as parameter is always either a PreparedRequest or a CachedRequest. Failing if the type is different will allow us to be aware of the subtlety if something else is ever used in the future